ui/src/dashboards/components/TableOptions.tsx

```diff
@@ -146,6 +154,8 @@ export class TableOptions extends PureComponent<Props, {}> {
       text: field.displayName || field.internalName,
     }))
 
+    const fields = fieldNames.length > 1 ? fieldNames : this.computedFieldNames
```

Comment:
I wonder- I haven't tried this out, and I'm not sure.. Could running computedFieldNames and updating fieldNames go in to the ShouldComponentUpdate function? As it is now, we would never re-updating fieldNames- right? Maybe we never need to.. hrmmm.. Anyway, consider.

Reply:
They get saved on the first time a field is updated (either their visibility or display name) and whenever the TableOptions is mounted. The only time its using this.computedFieldNames instead of fieldNames is when when a cell is first changed to a table.